dimitriosv/week3/app/Messages.jsx

```diff
@@ -0,0 +1,52 @@
+//import Message from './Message.jsx';
+var Message = require('./Message');
+
+var Messages = React.createClass({
+    render: function () {
+        
+        return  <div>
+                    <div>
+                        {this.props.messages.map(function(message,i) {
+                            if (message.isArchived) {
+                                ;
+                            } else {
```

Comment:
Invert this statement: `if (!message.isArchived)`

If I were you, I would first, `filter` and then `map`

```javscript 
this.props.messages
   .filter(function(message){ 
      return !message.isArchived;
   })
   .map(function(message){
      return /* JSX */ 
   })
```